Initial commit V1 - web personas smart

harshithsunku opened the pull request.

harshithsunku commented:
> 1/ it's functional btw, man could you clean your code before requesting for review ?
> 
> @harshithsunku

Hi @beastoin ,

Done, Kindle review.

beastoin commented:
1/ ok
2/ can you use the same firebase db as the current one ?
3/ the UI looks boring but tbh i don't know how to improve it haha, any idea ?

please note the step to deploy(release) such as, setup the Stripe account, add new pricing, add new env vars, e.t.c

@harshithsunku ^

@mdmohsin7 man, could you help testing this ticket ?

mdmohsin7 commented:
@beastoin tested with our current db and stripe test credentials. It works very well 👍

https://github.com/user-attachments/assets/0cb15dc3-8579-456e-a914-1f8955f89160

beastoin commented:
> 1/ ok 2/ can you use the same firebase db as the current one ? 3/ the UI looks boring but tbh i don't know how to improve it haha, any idea ?
> 
> please note the step to deploy(release) such as, setup the Stripe account, add new pricing, add new env vars, e.t.c
> 
> @harshithsunku ^
> 
> @mdmohsin7 man, could you help testing this ticket ?

@harshithsunku sry for annoying you (if any) btw could you check ? i really want your hard work to pay off.

harshithsunku commented:
> > 1/ ok 2/ can you use the same firebase db as the current one ? 3/ the UI looks boring but tbh i don't know how to improve it haha, any idea ?
> > please note the step to deploy(release) such as, setup the Stripe account, add new pricing, add new env vars, e.t.c
> > @harshithsunku ^
> > @mdmohsin7 man, could you help testing this ticket ?
> 
> @harshithsunku sry for annoying you (if any) btw could you check ? i really want your hard work to pay off.

Hi @beastoin ,

Apologies for the delayed response. I was held up with release activities and couldn't review this earlier.

Below is the deployment plan:

Environment Variables
Please add the following environment variables:

- NEXT_PUBLIC_STRIPE_PUBLISHABLE_KEY
- STRIPE_SECRET_KEY
- STRIPE_WEBHOOK_SECRET
- STRIPE_PRICE_PRO_MONTHLY
- FIREBASE_PROJECT_ID
- FIREBASE_CLIENT_EMAIL
- FIREBASE_PRIVATE_KEY

Stripe Configuration
1. Create a new product in Stripe.
2. Create a price for the product.
3. Replace the STRIPE_PRICE_PRO_MONTHLY value with the newly created price ID.

Webhook Events
Please add the following events for the webhook:

- checkout.session.completed
- invoice.payment_succeeded
- customer.subscription.updated
- customer.subscription.deleted

Let me know if you have any questions.

Regards,
Harshith

harshithsunku commented:
> 2/ could we use the current firebase config instead of new one ? we've already have firebase. https://github.com/BasedHardware/omi/blob/main/personas-open-source/Dockerfile#L13-L17
> 
> >FIREBASE_PROJECT_ID
> FIREBASE_CLIENT_EMAIL
> FIREBASE_PRIVATE_KEY
> 
> @harshithsunku 

Hi @beastoin ,

The newly added Firebase environment variables are for Firebase Admin and should only be used on the server side. The FIREBASE_CLIENT_EMAIL and FIREBASE_PRIVATE_KEY were not used before this PR. I can reuse NEXT_PUBLIC_FIREBASE_PROJECT_ID as FIREBASE_PROJECT_ID.

Regards,
Harshith.

harshithsunku commented:
Hi @beastoin ,

I have resolved conflicts and FIREBASE_PROJECT_ID with NEXT_PUBLIC_FIREBASE_PROJECT_ID.
Kindly review.

Regards,
Harshith.

beastoin commented:
Hey @harshithsunku, sorry for the late reply. By the way, we're considering the needs of this feature since we really want to keep the repo lean. We will get back to you soon.

harshithsunku commented:
> Hey @harshithsunku, sorry for the late reply. By the way, we're considering the needs of this feature since we really want to keep the repo lean. We will get back to you soon.

Hi @beastoin ,

Ok, Thanks.

Regards,
Harshith.

skywinder commented:
Well, two weeks later there was a major refactoring with many file renames, so it’s unclear how this can be merged now.

So the question is:
– To the author @harshithsunku  — are you willing to resolve the merge conflicts?
– To the team @beastoin  — if the author isn’t sure how to proceed, can someone step in to avoid leaving this as a dead pull request?

harshithsunku commented:
> Well, two weeks later there was a major refactoring with many file renames, so it’s unclear how this can be merged now.
> 
> So the question is: – To the author @harshithsunku — are you willing to resolve the merge conflicts? – To the team @beastoin — if the author isn’t sure how to proceed, can someone step in to avoid leaving this as a dead pull request?

Hi @skywinder ,

I’m happy to resolve the merge conflicts once @beastoin gives the go-ahead. Since he mentioned he’d get back, I’ve been waiting for confirmation before proceeding.

If I resolve the conflicts now and the PR still doesn’t get merged soon, there’s a risk that further changes in the codebase will cause new conflicts again.

Thanks!

beastoin commented:
@skywinder he is just waiting for me - you could find the related discussion on the discord https://discord.com/channels/1192313062041067520/1295877183587221646/1352726824177897492

@harshithsunku the results are not final yet but tbh, i think it's really hard to find the value in merging this PR into omi at this point. don't get me wrong, your code works well but the timing isn't right anymore since it's been months from the persona launch - and, the product has changed so much. so, let do it this way.

1/ close this PR
2/ feel free to reach out to me on https://discord.omi.me \@thinh to discuss more about how could we pay for your help.

alternatively, if @kodjima33 has any different perspective he's welcome to request for reopen the PR.

thank you ~
